modules/tinymce/src/core/main/ts/api/EditorUpload.ts

```diff
@@ -28,11 +41,60 @@ export interface UploadResult {
 }
 
 interface EditorUpload {
+  /**
+   * Cache of blob elements created in an editor instance.
+   *
+   * @property blobCache
+   * @type Object
+   */
   blobCache: BlobCache;
+
+  /**
+   * Adds a custom filter that controls the images which are included in the scan.
+   * Images must return true on every added filter to be considered valid.
+   *
+   * @method addFilter
+   * @param {Function} filter Function which filters each image upload.
+   * @example
+   * // Filter which images are uploaded.
+   * tinymce.activeEditor.EditorUpload.addFilter((image) => {
+   *   const maxSize = 1920 * 1080;
+   *   const imageSize = image.width * image.height;
+   *   return imageSize < maxSize;
+   * });
+   */
   addFilter: (filter: (img: HTMLImageElement) => boolean) => void;
+
+  /**
+   * Uploads all the data uri/blob uri images scanned from the editor content to the server.
+   *
+   * @method uploadImages
+   * @return {Promise} Promise instance with images and status for each image.
+   */
   uploadImages: () => Promise<UploadResult[]>;
+
+  /**
+   * Uploads all data uri/blob uri images to the server only when automatic uploads are enabled.
```

Comment:
Is it called automatically as well? if the logic for automatic uploading images is elsewhere, then this function looks like a duplicate of `uploadImages`

Reply:
When automatic uploads are enabled, this is called on `setContent`, so every time the content is changed. It is essentially a duplicate of `uploadImages` which only runs when automatic uploads are enabled, if you were to only describe the API and not it's use. In time I can see this API being removed as the conditional within the `editor.on('setContent')` call already checks if automatic uploads are enabled and could call `uploadImages` instead, and `uploadImagesAuto` should only be called then anyway.